core/species.cpp

```diff
@@ -3387,6 +3514,49 @@ void Species::AboutToSplitSubpop(void)
 	community_.tree_seq_tick_offset_ += 0.00001;
 }
 
+void Species::CopySharedTablesIn(tsk_table_collection_t &p_tables)
```

Comment:
Maybe this should be called "LinkSharedTablesIn" or something other than "copy"?

Reply:
Hm, or `LinkSharedTableColumns`?

Reply:
Wait, I'm confused here - isn't `p_tables.nodes` just a [pointer to a NodeTable](https://tskit.dev/tskit/docs/latest/c-api.html#c.tsk_table_collection_t)? And the [NodeTable object](https://tskit.dev/tskit/docs/latest/c-api.html#c.tsk_node_table_t) has it's own `num_rows`? From the description it sounds like you're reassigning the pointers to the separate columns of the node table, but that's not what's happening here?

Reply:
Not sure whether that clarifies things sufficiently or not...

Reply:
Ah, yep - sorry!